code/datums/components/gps.dm

```diff
@@ -108,8 +108,11 @@ GLOBAL_LIST_EMPTY(GPS_list)
 		return data
 
 	var/turf/curr = get_turf(parent)
+	var/datum/virtual_level/vlevel = SSmapping.get_virtual_level(curr)
+	var/datum/map_zone/mapzone = vlevel.parent_map_zone
+	var/list/coords = vlevel.get_relative_coords(curr)
 	data["currentArea"] = "[get_area_name(curr, TRUE)]"
-	data["currentCoords"] = "[curr.x], [curr.y], [curr.get_virtual_z_level()]"
+	data["currentCoords"] = "[coords[1]], [coords[2]], [mapzone.id], [vlevel.relative_id]"
```

Comment:
doesn't need a change, but THANK YOU FOR THIS

Reply:
<3